### PR TITLE
After confirm letter in form wizard

### PR DESCRIPTION
**Pull request by Radiansz**
Enabling and disabling after confirm letter for forms. Saving AC letter in form, removing AC letter from form.

### CODE REVIEW

File: src/blocks/templateEditor/templateEditor.js [hunk 1]
```diff
@@ -73,9 +73,14 @@ define(function(require) {
...
                 var currentSidebar = self.models.editor.get('currentSidebar');
                 self.showSidebarByMark('.' + currentSidebar.mark, currentSidebar.animate, currentSidebar.options);
-            })
+            });
 
-   
+            if(!this.models.template) {
+                this.models.template = (new(require('resources/presentation/draft/block/model'))({
+                    'json': {},
+                    'isTemp': true
```
Comment (zouxuoz) [inline src/blocks/templateEditor/templateEditor.js]: Where it is used?

Comment (Radiansz) [inline src/blocks/templateEditor/templateEditor.js]: Removed

File: src/blocks/form/form.less [hunk 2]
```diff
@@ -652,3 +652,7 @@
 .form__group:not(:last-child) {
     margin-bottom: 20px;
 }
+
+.form.form_hidden {
```
Comment (rosendi) [inline src/blocks/form/form.less]: ???

File: src/blocks/form/formAfterFillSettings/formAfterFillSettings.less [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+	display: none;
+}
+
+.form_formAfterFillSettings.form_withletter .form__letter {
+	display: block;
+}
+
+.form_formAfterFillSettings .form__text {
+	padding-bottom: 20px;
+	width: 600px;
```
Comment (rosendi) [inline src/blocks/form/formAfterFillSettings/formAfterFillSettings.less]: Use `max-width` here

File: src/blocks/form/formAfterFillSettings/formAfterFillSettings.less [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+}
+
+.form_formAfterFillSettings .inputText_length_medium {
+	width: 500px;
```
Comment (rosendi) [inline src/blocks/form/formAfterFillSettings/formAfterFillSettings.less]: Use `max-width` here

File: src/blocks/form/formAfterFillSettings/formAfterFillSettings.css [hunk 5]
```diff
@@ -0,0 +1,13 @@
+.form_formAfterFillSettings .form__letter {
+  display: none;
+}
+.form_formAfterFillSettings.form_withletter .form__letter {
```
Comment (rosendi) [inline src/blocks/form/formAfterFillSettings/formAfterFillSettings.css]: Camelize: form_withLetter

File: src/blocks/form/formAfterFillSettings/formAfterFillSettings.js [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+    return Form.extend({
+        template: require('tpl!./formAfterFillSettings.ejs'),
+
+        defaults: {
+            modifiers: {
+                flexible: true,
+                formAfterFillSettings: true
+            },
+            linkValidators: {
+                'fillUrl': [require('kit/validators/required'), require('kit/validators/url')()]
```
Comment (rosendi) [inline src/blocks/form/formAfterFillSettings/formAfterFillSettings.js]: Move validator's requiring to the top.

Comment (rosendi) [inline src/blocks/form/formAfterFillSettings/formAfterFillSettings.js]: Ex: `RequiredValidator`

File: src/blocks/form/formAfterFillSettings/formAfterFillSettings.js [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+            },
+            letterValidators: {
+                'fromName': require('kit/validators/required'),
+                'fromEmail': [require('kit/validators/required'), require('kit/validators/email'), require('kit/validators/DMARC')],
+                'subject': require('kit/validators/required')
+            }
+        },
+
+        events: {
+            'change .js-toggle-form input': function(event) {
```
Comment (rosendi) [inline src/blocks/form/formAfterFillSettings/formAfterFillSettings.js]: Move it to a private function.

File: src/blocks/form/formAfterFillSettings/formAfterFillSettings.js [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+                this.models.form.set('welcomeDraft', this.models.draft);
+            } else {
+                this.models.form.unset('welcomeDraft');
+            }
+
+            this.models.form.get('fillLink').set('url', data.fillUrl);
+
+            return this.models.form.save({}, {
+                silent: true,
+                forse: this.forseSave
```
Comment (rosendi) [inline src/blocks/form/formAfterFillSettings/formAfterFillSettings.js]: Looks like it's a bug `forSe` and `forSeSave`.

File: src/blocks/form/formEditLetter/formEditLetter.js [hunk 9]
```diff
@@ -11,14 +11,14 @@ define(function(require) {
                 flexible: true
             }
         },
-
         validators: {
```
Comment (rosendi) [inline src/blocks/form/formEditLetter/formEditLetter.js]: Could you require validators in the top here tooooo..,

File: src/blocks/wizard/__info/formEditLetter/formEditLetter.js [hunk 10]
```diff
@@ -3,16 +3,21 @@ define(function(require) {
     var Info = require('blocks/wizard/__info/__info');
 
     return Info.extend({
-        template: require('tpl!./formEditConfirmLetter.ejs'),
+        template: require('tpl!./formEditLetter.ejs'),
+        defaults: {
```
Comment (rosendi) [inline src/blocks/wizard/__info/formEditLetter/formEditLetter.js]: Add a new line.

File: src/blocks/wizard/__info/formEditLetter/formEditLetter.js [hunk 11]
```diff
@@ -3,16 +3,21 @@ define(function(require) {
...
             if (!this.models.form.get('fillDraft')) {
                 return;
             }
 
             var self = this,
-                letter = this.$el.find('.js-preview');
-
+                letter = this.$el.find('.js-preview'),
+                draft = self.models.form.get(self.draftKey);
+            if(!draft)
```
Comment (rosendi) [inline src/blocks/wizard/__info/formEditLetter/formEditLetter.js]: Wrap it into brackets. Never use this style.

```
if (!draft) {
  ...
}
```

File: src/blocks/wizard/__step/formEditConfirmLetter/formEditConfirmLetter.ejs [hunk 12]
```diff
@@ -13,7 +13,7 @@
         <div class="wizard__stepTitleText"><%- self.title %></div>
         <div class="wizard__stepDoneTitleText"><%- self.doneTitle %></div>
         <div class="wizard__stepTitleInfo">
-            <%= self.include(require('blocks/wizard/__info/formEditConfirmLetter/formEditConfirmLetter'), {
+            <%= self.include(require('blocks/wizard/__info/formEditLetter/formEditLetter'), {
```
Comment (rosendi) [inline src/blocks/wizard/__step/formEditConfirmLetter/formEditConfirmLetter.ejs]: Don't use inline require.

File: src/blocks/wizard/__step/formEditWelcomeLetter/formEditWelcomeLetter.ejs [hunk 13]
```diff
@@ -13,24 +13,22 @@
         <div class="wizard__stepTitleText"><%- self.title %></div>
         <div class="wizard__stepDoneTitleText"><%- self.doneTitle %></div>
         <div class="wizard__stepTitleInfo">
-            <%= self.include(require('blocks/wizard/__info/formEditWelcomeLetter/formEditWelcomeLetter'), {
+            <%= self.include(require('blocks/wizard/__info/formEditLetter/formEditLetter'), {
```
Comment (rosendi) [inline src/blocks/wizard/__step/formEditWelcomeLetter/formEditWelcomeLetter.ejs]: Don't use inline require.

File: src/resources/form/model.js [hunk 14]
```diff
@@ -22,67 +23,72 @@ define(function(require) {
...
+        createLink: function() {
+            var link = new Link();
+            return link;
         },
 
         destroyDraft: function(key) {
             var draft = this.get(key);
 
-            if (draft.destroy) {
+            if(draft && draft.destroy && draft.get('id')) {
```
Comment (rosendi) [inline src/resources/form/model.js]: Is it reduntant? If a draft has an id, have we check for `destroy` presence?

File: src/resources/form/model.js [hunk 15]
```diff
@@ -175,9 +175,9 @@ define(function(require) {
             }
         },
 
-        fetchFillLink: function() {
-            if (this.get('fillLink')) {
-                return this.get('fillLink').fetch();
+        fetchLink: function(key) {
+            if (this.get(key)) {
```
Comment (rosendi) [inline src/resources/form/model.js]: Wrap into brackets.

File: src/resources/link/model.js [hunk 16]
```diff
@@ -4,7 +4,29 @@ define(function(require) {
         api = require('kit/api/api');
 
     return Model.extend({
-        rootAction: 'link'
+        rootAction: 'link',
+
+        save: function(options) {
+    		var model = this;
+            return api.call(_.extend({
+                action: model.get('id') ? 'link.set' : 'link.create',
```
Comment (rosendi) [inline src/resources/link/model.js]: It's better to move into a private function `_actionName` or `_getActionName`.

File: src/blocks/form/formAfterFillSettings/formAfterFillSettings.js (hunk 7 above)
Comment (rosendi) [inline src/blocks/form/formAfterFillSettings/formAfterFillSettings.js]: Also I don't like `.js-toggle-form`. May be `.js-toggle-letter`?

File: src/blocks/wizard/__step/formEditWelcomeLetter/formEditWelcomeLetter.ejs [hunk 17]
```diff
@@ -13,24 +13,22 @@
...
+                collections: self.collections,
+                draftKey: 'welcomeDraft'
             }) %>
         </div>
     </div>
     <div class="wizard__stepInner">
+
         <div class="wizard__stepContent">
-            <%= self.include(require('blocks/form/formEditLetter/formEditLetter'), {
+            <%= self.include(require('blocks/form/formAfterFillSettings/formAfterFillSettings'), {
```
Comment (rosendi) [inline src/blocks/wizard/__step/formEditWelcomeLetter/formEditWelcomeLetter.ejs]: Don't use inline require.

File: src/blocks/form/formAfterFillSettings/formAfterFillSettings.js (hunk 8 above)
Comment (Radiansz) [inline src/blocks/form/formAfterFillSettings/formAfterFillSettings.js]: I didnt get that one

File: src/resources/form/model.js [hunk 18]
```diff
@@ -123,22 +136,9 @@ define(function(require) {
...
-        },
-
-        needToFetchFillDraft: function() {
-            return this.fetchedDependencies.indexOf('fillDraft') !== -1;
-        },
-
-        needToFetchFillLink: function() {
-            return this.fetchedDependencies.indexOf('fillLink') !== -1;
+        fetchedDependencies: ['questionnaire', 'presentation', 'fillDraft', 'fillLink', 'welcomeDraft', 'welcomeLink', 'sourceLink'],
+        needToFetch: function(entityName) {
```
Comment (rosendi) [inline src/resources/form/model.js]: Add a new line above.

File: src/resources/form/model.js [hunk 19]
```diff
@@ -225,22 +225,15 @@ define(function(require) {
...
-                });
-            }
+            this.parseDraft(response, 'fill', data);
+            this.parseDraft(response, 'welcome', data);
 
-            if (!this.get('fillLink') && response.obj.fill && response.obj.fill.link) {
-                data['fillLink'] = new LinkModel({
-                    id: response.obj.fill.link
-                });
+            if(!this.get('fillLink') && response.obj.fill &&  response.obj.fill.link) {
```
Comment (rosendi) [inline src/resources/form/model.js]: Add space after `if`.

File: src/resources/form/model.js [hunk 20]
```diff
@@ -225,22 +225,15 @@ define(function(require) {
...
+                data['fillLink'] = new Link({ id: response.obj.fill.link });
             }
 
-            if (!this.get('welcomeDraft') && response.obj.welcome && response.obj.welcome.draft) {
-                data['welcomeDraft'] = new DraftModel({
-                    id: response.obj.welcome.draft
-                });
+            if(!this.get('welcomeLink') && response.obj.welcome &&  response.obj.welcome.link) {
```
Comment (rosendi) [inline src/resources/form/model.js]: Add space after `if`.

File: src/blocks/form/formAfterFillSettings/formAfterFillSettings.ejs [hunk 21]
```diff
@@ -0,0 +1,70 @@
...
+    var link = this.models.form.get('fillLink');
+%>
+
+<form class="<%- className %>">
+    <div class="form__body">
+        <div class="form__text">
+            <%= i18n('После того, как подписчик подтвердит форму, он будет направлен на страницу "Спасибо за подтверждение формы". Вы можете указать другой URL и показать подписчику свой вариант страницы благодарности.') %>
+        </div>
+        <div class="form__field">
+            <%= self.include(UI.inputText, {
```
Comment (rosendi) [inline src/blocks/form/formAfterFillSettings/formAfterFillSettings.ejs]: Omit UI.

File: src/blocks/form/formAfterFillSettings/formAfterFillSettings.ejs [hunk 22]
```diff
@@ -0,0 +1,70 @@
...
+                    modifiers: {
+                        length: 'medium'
+                    }
+                }) %>
+        </div>
+        <div class="form__text">
+            <%= i18n('После подтверждения формы можно выслать письмо благодарности') %>
+        </div>
+        <div class="form__field">
+            <%= this.include(UI.checkbox, {
```
Comment (rosendi) [inline src/blocks/form/formAfterFillSettings/formAfterFillSettings.ejs]: Omit UI.

File: src/blocks/form/formAfterFillSettings/formAfterFillSettings.ejs [hunk 23]
```diff
@@ -0,0 +1,70 @@
...
+                    mark: 'toggle-letter',
+                    checked: this.models.form.get('welcomeDraft'),
+                    label: i18n('Отправить письмо благодарности')
+                }) %>
+        </div>
+        <div class="form__letter">
+            <div class="form__field">
+                <%= self.include(UI.inputText, {
```
Comment (rosendi) [inline src/blocks/form/formAfterFillSettings/formAfterFillSettings.ejs]: Omit UI.

File: src/blocks/form/formAfterFillSettings/formAfterFillSettings.ejs [hunk 24]
```diff
@@ -0,0 +1,70 @@
...
+                    label: i18n('Имя отправителя'),
+                    name: 'fromName',
+                    value: self.models.draft.get('letter.from@name')
+                }) %>
+            </div>
+
+            <div class="form__field">
+                <%= self.include(UI.inputText, {
```
Comment (rosendi) [inline src/blocks/form/formAfterFillSettings/formAfterFillSettings.ejs]: Omit UI.

File: src/blocks/form/formAfterFillSettings/formAfterFillSettings.ejs [hunk 25]
```diff
@@ -0,0 +1,70 @@
...
+                    label: i18n('Email отправителя'),
+                    name: 'fromEmail',
+                    value: self.models.draft.get('letter.from@email')
+                }) %>
+            </div>
+
+            <div class="form__field">
+                <%= self.include(UI.inputText, {
```
Comment (rosendi) [inline src/blocks/form/formAfterFillSettings/formAfterFillSettings.ejs]: Omit UI.

File: src/blocks/form/formAfterFillSettings/formAfterFillSettings.js [hunk 26]
```diff
@@ -0,0 +1,95 @@
...
+            }
+            if (!this.models.form.get('fillLink')) {
+                this.models.form.set('fillLink', this.models.form.createLink());
+            }
+
+            return Form.prototype.initialize.apply(this, arguments);
+        },
+
+        save: function(data) {
+            if (this.modifiers.withLetter) {
```
Comment (rosendi) [inline src/blocks/form/formAfterFillSettings/formAfterFillSettings.js]: !

File: src/nls/en-us/dictionary.js [hunk 27]
```diff
@@ -1739,6 +1739,13 @@ define(function(require) {
...
         'Скопировать код': 'Copy code',
-        'Код формы скопирован в буфер обмена': 'The embed code has been copied to clipboard'
+        'Код формы скопирован в буфер обмена': 'The embed code has been copied to clipboard',
+        'После того, как подписчик подтвердит форму, он будет направлен на страницу "Спасибо за подтверждение формы". Вы можете указать другой URL и показать подписчику свой вариант страницы благодарности.': 'After a subscriber confirms the form, he will be redirected to the page "Thank you for the confirmation of form". You can specify another URK, and show your version of thank you page to a subscriber.',
+        'Адрес страницы благодарности': 'Thank you page URL', 
+        'После подтверждения формы можно выслать письмо благодарности': 'After confirmation of the form you can send thank you email',
+        'Отправить письмо благодарности': 'Send thank you letter',
```
Comment (rosendi) [inline src/nls/en-us/dictionary.js]: Send a thank you letter

**Comment by rosendi** [comment]
LGTM

**Comment by rosendi** [comment]
LGTM

**Pull request merged**